crates/rpc/rpc-eth-api/src/api/servers/server.rs

```diff
@@ -1,49 +1,37 @@
-//! Implementation of the [`jsonrpsee`] generated [`reth_rpc_api::EthApiServer`] trait
-//! Handles RPC requests for the `eth_` namespace.
+//! Implementation of the [`jsonrpsee`] generated [`EthApiServer`] trait. Handles RPC requests for
+//! the `eth_` namespace.
 
-use super::EthApiSpec;
-use crate::{
-    eth::{
-        api::{EthApi, EthTransactions},
-        error::EthApiError,
-    },
-    result::{internal_rpc_err, ToRpcResult},
-};
 use alloy_dyn_abi::TypedData;
 use jsonrpsee::core::RpcResult as Result;
-use reth_evm::ConfigureEvm;
-use reth_network_api::NetworkInfo;
 use reth_primitives::{Address, BlockId, BlockNumberOrTag, Bytes, B256, B64, U256, U64};
-use reth_provider::{
-    BlockIdReader, BlockReader, BlockReaderIdExt, ChainSpecProvider, EvmEnvProvider,
-    HeaderProvider, StateProviderFactory,
-};
-use reth_rpc_api::EthApiServer;
 use reth_rpc_types::{
     serde_helpers::JsonStorageKey,
     state::{EvmOverrides, StateOverride},
     AccessListWithGasUsed, AnyTransactionReceipt, BlockOverrides, Bundle,
     EIP1186AccountProofResponse, EthCallResponse, FeeHistory, Header, Index, RichBlock,
-    StateContext, SyncStatus, TransactionRequest, Work,
+    StateContext, SyncStatus, Transaction, TransactionRequest, Work,
 };
-use reth_transaction_pool::TransactionPool;
 use tracing::trace;
 
+use crate::{
+    result::internal_rpc_err,
+    servers::{
+        EthApiSpec, EthBlocks, EthCall, EthFees, EthState, EthTransactions, LoadReceipt, Trace,
+    },
+    EthApiError, EthApiServer, ToRpcResult,
+};
+
 #[async_trait::async_trait]
-impl<Provider, Pool, Network, EvmConfig> EthApiServer for EthApi<Provider, Pool, Network, EvmConfig>
+impl<T> EthApiServer for T
 where
-    Self: EthApiSpec + EthTransactions,
-    Pool: TransactionPool + 'static,
-    Provider: BlockReader
-        + BlockIdReader
-        + BlockReaderIdExt
-        + ChainSpecProvider
-        + HeaderProvider
-        + StateProviderFactory
-        + EvmEnvProvider
-        + 'static,
-    Network: NetworkInfo + Send + Sync + 'static,
-    EvmConfig: ConfigureEvm + 'static,
+    Self: EthApiSpec
+        + EthTransactions
+        + EthBlocks
+        + EthState
+        + EthCall
+        + EthFees
+        + Trace
+        + LoadReceipt,
```

Comment:
this is very neat.

this also doesn't prevent other impls from implementing it manually and or reuse the existing functions.

the tier 2 endpoints (trace,debug) still operate on the subtraits, which is fine